src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/avoidoutdatedusage/InputAvoidOutdatedUsageForCollectors.java

```diff
@@ -0,0 +1,43 @@
+/*
+AvoidOutdatedUsage
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.avoidoutdatedusage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+
+public class InputAvoidOutdatedUsageForCollectors {
+
+    static String FOO = "foo";
+
+    private static void Collectors() {
+        // TODO Usage of API documented as @since 16+
+        // Stream.of(FOO).toList(); // ok, as modern
+        Stream.of(FOO).collect(
+        Collectors.toList()); // violation, Avoid ''.collect(Collectors.toList()) -> .toList()''.
+        // Can be replaced with 'java.util.ArrayList' constructor
+        new ArrayList<>(List.of(FOO)); // ok, as modern
+        Collector<Object, ?, List<Object>> list =
+            Collectors.toList(); // violation, Avoid ''.collect(Collectors.toList()) -> .toList()''.
+        toList(); // violation, Avoid ''.collect(Collectors.toList()) -> .toList()''.
+        // TODO Usage of API documented as @since 16+
+        // List.of(FOO).stream().toList(); // ok, as modern
+        // Stream.of(FOO).toList(); // ok, as modern
+        // Test other Collector methods
+        Collectors.toCollection(ArrayList::new); // ok, modern API
+        // FIXME: but was       : [22, 23, 26, 27, 28, 35]
+        // Foo.toList(); // ok, as custom
```

Comment:
> No we are not java types aware, no class path details. All we have are strings in AST. https://checkstyle.org/writingchecks.html#Limitations

can we fix this overloading issue then? We only should exclude class specific. @romani

Reply:
No types available, we can not distinguish overloaded methods.
We don't know hierarchy of types.

All this limitations makes any implementation to be very error prone.

Reply:
yes thats right. Maybe then go like [lombok ](https://projectlombok.org/features/experimental/UtilityClass) ? This always worked for my and better to have it and not miss it. We can tell as it is. 

Just check out codebase its capable of so we are example for what its good for. 

`in the land of the blind, the one-eyed man is king.`